### PR TITLE
CRM_Utils_Date - Get month names from IntlDateFormatter. Ignore timezones.

### PR DESCRIPTION
**Pull request by totten**
Overview
----------------------------------------

Alternative to #25066. Not tested. Let's see what CI says...

Before
----------------------------------------
_What is the old user-interface or technical-contract (as appropriate)?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

After
----------------------------------------
_What changed? What is new old user-interface or technical-contract?_
_For optimal clarity, include a concrete example such as a screenshot, GIF ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)), or code-snippet._

Technical Details
----------------------------------------
_If the PR involves technical details/changes/considerations which would not be manifest to a casual developer skimming the above sections, please describe the details here._

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

### CODE REVIEW

**Comment by totten** [comment]
The first test-run showed that `ts('March')` and `IntlDateFormatter` produce different capitalization for month-names in `es_MX`.

```diff
 Array (
     0 => 'ene.'
     1 => 'mar.'
-    2 => 'Marzo'
+    2 => 'marzo'
     3 => 'jueves'
 )
```
```diff
-'Sí No Abril'
+'Sí No abril'
```

As noted in [c158e23](https://github.com/civicrm/civicrm-core/pull/25154/commits/c158e23e2453a7e852fc234a47eec925097efab5), it seems like lowercase is probably more correct.

File: CRM/Utils/Date.php [hunk 1]
```diff
@@ -232,21 +232,16 @@ public static function getFullWeekdayNames() {
...
-        8 => $intl_formatter->format(strtotime('1 August')),
-        9 => $intl_formatter->format(strtotime('1 September')),
-        10 => $intl_formatter->format(strtotime('1 October')),
-        11 => $intl_formatter->format(strtotime('1 November')),
-        12 => $intl_formatter->format(strtotime('1 December')),
-      ];
+      // Note: IntlDateFormatter provides even more strings than `strftime()` or `l10n/*/civicrm.mo`.
+      // Note: Consistently use UTC for all requests in resolving these names. Avoid edge-cases where TZ support is inconsistent.
+      $intlFormatter = IntlDateFormatter::create(CRM_Core_I18n::getLocale(), IntlDateFormatter::MEDIUM, IntlDateFormatter::MEDIUM, 'UTC', IntlDateFormatter::GREGORIAN, 'MMM');
+      $monthNums = range(1, 12);
```
Comment (demeritcowboy) [inline CRM/Utils/Date.php:238]: I didn't even know such a function existed.

Overall this makes sense. I think using `intl` is the way to go, even though it means more code.

**Comment by demeritcowboy** [comment]
Tested with a couple timezones and also italian.

By the way I now believe this is a bug in ICU that was fixed around version 68. I can't point to a specific changelog item but the reason I think this is that at first I couldn't reproduce locally anymore, even though I could before. In my windows php, the version of ICU matches up to php like so: php 7.4 => 66, php 8.0 => 68, php 8.1 => 70. I can only reproduce in php 7.4. On dmaster, it looks like php 7.4 => 65, and on the test nodes php 8.1 => 64 (which is weird). So this would explain why the problem is reproducible on both dmaster and test nodes, but I can only reproduce on 7.4.

Anyway, this method seems to work on both.

**Comment by mlutfy** [comment]
@demeritcowboy @totten Thank you!

**Pull request merged**